tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Core.cs

```diff
@@ -16,38 +16,73 @@ namespace Datadog.Trace.Util.Http
     internal static partial class HttpRequestExtensions
     {
         private const string NoHostSpecified = "UNKNOWN_HOST";
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(HttpRequestExtensions));
 
         internal static Dictionary<string, object> PrepareArgsForWaf(this HttpRequest request, RouteData routeDatas = null)
         {
             var url = GetUrl(request);
-            var headersDic = new Dictionary<string, string>(request.Headers.Keys.Count);
+            var headersDic = new Dictionary<string, string[]>(request.Headers.Keys.Count);
             foreach (var k in request.Headers.Keys)
             {
                 if (!k.Equals("cookie", System.StringComparison.OrdinalIgnoreCase))
                 {
-                    headersDic.Add(k.ToLowerInvariant(), request.Headers[k]);
+                    var key = k.ToLowerInvariant();
+                    if (!headersDic.ContainsKey(key))
+                    {
+                        headersDic.Add(key, request.Headers[key]);
+                    }
+                    else
+                    {
+                        Log.Warning("Header {key} couldn't be added as argument to the waf", key);
+                    }
```

Comment:
Probably not worth it, but you _could_ specialise this (and the querystring dic) for .NET Core and avoid two lookups. Probably not worth it though, as`TryAdd` isn't in netstandard 2.0
```suggestion
#if NETCOREAPP
                    if (!headersDic.TryAdd(key, request.Headers[key]))
                    {
                        Log.Warning("Header {key} couldn't be added as argument to the waf", key);
                    }
#else                    
                    if (!headersDic.ContainsKey(key))
                    {
                        headersDic.Add(key, request.Headers[key]);
                    }
                    else
                    {
                        Log.Warning("Header {key} couldn't be added as argument to the waf", key);
                    }
#endif
```

Reply:
ok! added in 12edbe0